rustls/src/crypto/signer.rs

```diff
@@ -59,6 +60,12 @@ pub trait SigningKey: Debug + Send + Sync {
     /// using the chosen scheme.
     fn choose_scheme(&self, offered: &[SignatureScheme]) -> Option<Box<dyn Signer>>;
 
+    /// Get the RFC 5280-compliant SubjectPublicKeyInfo (SPKI) of this [`SigningKey`] if available.
+    fn public_key(&self) -> Option<SubjectPublicKeyInfoDer> {
```

Comment:
Should have a lifetime here, too, I guess we can do `<'_>` so implementations can decide to borrow from `self`.

Reply:
Done!